### PR TITLE
CLI: Support dumping the TX message in sign-only mode

### LINKED ISSUE #15786: cli: Add a flag for outputting raw transaction messages to be signed offline
#### Problem
The CLI does not allow creating a raw message that can be sent to a remote signer service for validation and signing. This makes it difficult to integrate into a generic signer service which simply validates and signs messages. The current CLI commands require a signer service to construct a transaction itself which adds extra complexity to a remote signer.

#### Proposed Solution
- Add `--dump-message` (naming?) flag to CLI commands which construct transactions. The cli should output the constructed tx message as base64 (configurable?) as well as the used recent blockhash.

After the message is signed, the signature(s) and recent blockhash should be supplied to the same CLI command according to the existing offline signing flow.

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

No way to output the binary transaction message with CLI

#### Summary of Changes

Add a flag to output the base64-encoded transaction message in sign-only mode

Fixes #15786

### CODE REVIEW

File: clap-utils/src/offline.rs [hunk 1]
```diff
@@ -19,6 +19,12 @@ pub const SIGNER_ARG: ArgConstant<'static> = ArgConstant {
     help: "Provide a public-key/signature pair for the transaction",
 };
 
+pub const DUMP_TRANSACTION_MESSAGE: ArgConstant<'static> = ArgConstant {
+    name: "dump_transaction_message",
+    long: "dump-transaction-message",
+    help: "Display the base64 encoded binary transaction message in sign-only mode",
```
Comment (jstarry) [inline clap-utils/src/offline.rs:25]: So you can only dump the message if you're signing? Is it possible to dump a message without signing?

Comment (t-nelson) [reply clap-utils/src/offline.rs:25]: Signing is optional in sign-only mode :upside_down_face: 
```
solana --keypair 11111111111111111111111111111111 transfer \
--sign-only --blockhash 11111111111111111111111111111111 \
--dump-transaction-message \
11111111111111111111111111111111 1 

Blockhash: 11111111111111111111111111111111
Transaction Message: AQAAAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAAIAAAwCAAAAAMqaOwAAAAA=
Absent Signers (Pubkey):
 1111111111111111111111111111111
```

File: cli-output/src/cli_output.rs [hunk 2]
```diff
@@ -2055,7 +2068,7 @@ mod tests {
         let signers = vec![present.as_ref(), absent.as_ref(), bad.as_ref()];
         let blockhash = Hash::new(&[7u8; 32]);
         tx.try_partial_sign(&signers, blockhash).unwrap();
-        let res = return_signers(&tx, &OutputFormat::JsonCompact).unwrap();
+        let res = return_signers(&tx, false, &OutputFormat::JsonCompact).unwrap();
```
Comment (CriesofCarrots) [inline cli-output/src/cli_output.rs:2071]: Worth one `dump_transaction_message: true` test case with expected golden?

**Review by CriesofCarrots — DISMISSED** [review]
lgtm

**Comment by CriesofCarrots** [comment]
How do you want to handle updating spl-token-cli?

File: cli-output/src/cli_output.rs (hunk 2 above)
Comment (t-nelson) [reply cli-output/src/cli_output.rs:2072]: Added [here](https://github.com/solana-labs/solana/pull/15834/commits/0030b47ac82167dee0861a8a9edd6c112781c962#diff-8dd480d7d15e5fcb41503792ea2dddbc345eb8efed94c73aadf89b7db7443a49R2080-R2092)!

**Comment by t-nelson** [comment]
> How do you want to handle updating spl-token-cli?

I'll do it once 1.5.15 crates are live.  So much less plumbing there.  The itch to refactor CLI similarly is getting strong now

**Comment by CriesofCarrots** [comment]
> I'll do it once 1.5.15 crates are live. So much less plumbing there. The itch to refactor CLI similarly is getting strong now

Ya, just don't think this will pass downstream-projects due to the `return_signers()` signature change

**Comment by t-nelson** [comment]
> > I'll do it once 1.5.15 crates are live. So much less plumbing there. The itch to refactor CLI similarly is getting strong now
> 
> Ya, just don't think this will pass downstream-projects due to the `return_signers()` signature change

Countdown to SPL in monorepo...

**Comment by t-nelson** [comment]
> Ya, just don't think this will pass downstream-projects due to the `return_signers()` signature change

I'm adding a `return_signers_with_config()`, `return_signers()` will just wrap it with defaults.  Re-plumbing CLI now :face_with_head_bandage:

File: clap-utils/src/offline.rs (hunk 1 above)
Comment (jstarry) [reply clap-utils/src/offline.rs:25]: If it's _only_ signing, and you're _not_ signing, is it really doing anything? 😝

Comment (t-nelson) [reply clap-utils/src/offline.rs:25]: `--sign-only` is in its pupal stage and will soon emerge as an adult `--offline` :wink:

**Pull request merged**